AdaptiveRoads/UI/QuayRoads/QuayRoadsPanelField.cs

```diff
@@ -0,0 +1,111 @@
+using AdaptiveRoads.Manager;
+using ColossalFramework.UI;
+using KianCommons;
+using KianCommons.UI;
+using static KianCommons.EnumBitMaskExtensions;
+using static KianCommons.EnumerationExtensions;
+using static KianCommons.ReflectionHelpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using AdaptiveRoads.Data.QuayRoads;
+using UnityEngine;
+using static UnityEngine.Object;
+using System.Globalization;
+using ColossalFramework;
+using AdaptiveRoads.Util;
+
+namespace AdaptiveRoads.UI.QuayRoads {
+    class QuayRoadsPanelField {
+        private NetInfo netInfo_;
+        private int sectionIndex_;
+        private FieldInfo fieldInfo_;
+        private IConvertible? flag_;
+        private RoadEditorPanel parentPanel_;
+
+        public QuayRoadsPanelField(NetInfo netInfo, int sectionIndex, FieldInfo fieldInfo, IConvertible flag, UIPanel parent, RoadEditorPanel parentPanel) {
+            netInfo_ = netInfo;
+            sectionIndex_ = sectionIndex;
+            fieldInfo_ = fieldInfo;
+            flag_ = flag;
+            parentPanel_ = parentPanel;
+            if (flag_ is not null) {
+
+                Log.Debug(flag.ToString());
+                Log.Debug(flag.ToUInt64().ToString());
+                var propertyCheckbox = parent.AddUIComponent<UICheckBoxExt>();
+                propertyCheckbox.Label = "";
+                propertyCheckbox.width = 20f;
+                propertyCheckbox.isChecked = (assetValue_ as IConvertible).IsFlagSet(flag);
+                propertyCheckbox.eventCheckChanged += (_, _isChecked) => {
+                    //TODO: find a better / more robust way to do this.
```

Comment:
find a better way to do this and move this to KianCommmons
(probably as SetFlag(ref enum obj, enum flag) or sth)

Reply:
```
                             AssetValue = (AssetValue as IConvertible).ToInt64() | flag.ToInt64();
```
you probably didn't notice but the reason why this line does not uses `CultureInfo.InvariantCulture` like the block above it is because you are using one of my extensions which totally handles every integer size/sign.

in any case `ToInt64(CultureInfo.InvariantCulture)` does work for 32bit integer too even without my extension.

Reply:
```
IConvirtable SetFlag(IConvirtable flags, IConvirtable flag, bool on);
```
the output should be same integer as input. a big switch case for every integer type would be required.
or maybe we can do everything in Int64 and then somehow cast it to the right integer using reflection.